lib/FIR/statement.def

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
As we discussed last week, there are other ways of implementing the definitions of the various statement data structures besides including a source file with various definitions of preprocessor macros that are going to be more readable and maintainable in the very long term.  Some combination of templates, inheritance, and singly-defined macros for boilerplate will likely replace the mechanism that you have here.  I am okay with approving this PR with this file in it, but one of us is going to have to replace it soon, I think.